test/custom-parser.test.js

```diff
@@ -1214,3 +1214,43 @@ test('route bodyLimit should take precedence over a custom parser bodyLimit', t
     })
   })
 })
+
+test('should be able to default parser for extra content type ', t => {
+  t.plan(4)
+  const fastify = Fastify()
+  t.tearDown(() => fastify.close())
+
+  fastify.post('/', { bodyLimit: 5 }, (request, reply) => {
+    reply.send(request.body)
+  })
+
+  fastify.addContentTypeParser('text/json', (req, done) => {
+    var body = ''
+    req.on('data', function onData (chunk) {
+      body += chunk
+    })
+    req.on('end', onEnd)
+    function onEnd () {
+      fastify.defaultJSONParser(req, body, done)
+      done()
+    }
+  })
```

Comment:
I think this should be:
```suggestion
  fastify.addContentTypeParser('text/json',fastify.defaultJSONParser)
```

Reply:
Unfortunately the `body `is of type Incoming message so if I use `fastify.addContentTypeParser('text/json',fastify.defaultJSONParser)` the secure-json parse would fail.

Reply:
🙁 I see. Yeah, I'm not sure what to do here. Having to redefine the handler every time is not very useful.